File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int [] ans = new int[2];
        for(int i = 0; i < nums.length; i ++){
            int complement = target - nums[i];
            if(map.containsKey(complement)){
                ans[1] = i;
                ans[0] = map.get(complement);
            }
            else{
                map.put(nums[i], i);
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        frequencies = {}
        for num in nums:
            if num not in frequencies:
                frequencies[num] = 1
            else:
                frequencies[num] += 1
        
        for key in frequencies:
            if frequencies[key] > 1:
                return True
        return False